### PR TITLE
Data/URI: add some convenience functions

### PR DESCRIPTION
**Pull request by nhaagen**
With a URI, I often want to add/change a parameter in the query and then use the URI as string.
This PR adds those functions to the data-type.

### CODE REVIEW

**Review by klees — CHANGES_REQUESTED** [review]
Hi @nhaagen,

thanks a lot! This LGTM in general but as usual I have some questions/suggestions:

* [ ] Is there a reason that we use `toString` as method name instead of the magic `__toString`?
* [ ] Let's explicitely declare `$params = []` [here](https://github.com/ILIAS-eLearning/ILIAS/pull/2864/files#diff-a331b8ead29f905062d29eb549e7dee1R402)

Please fix and then ping me immediately, this should make the table code simpler...

Best regards!

File: src/Data/URI.php [hunk 1]
```diff
@@ -381,4 +381,56 @@ public function getBaseURI() : string
...
+        $params = [];
+        parse_str($this->getQuery(), $params);
+        return $params;
+    }
+
+    /**
+     * Get the value of the given parameter (or null)
+     * @return  mixed|null
+     */
+    public function getParameter(string $param)
```
Comment (mjansenDatabay) [inline src/Data/URI.php:416]: You could use `:? string` as explicit return type here.

**Comment by nhaagen** [comment]
Hi @klees ,
thanks for your comments - I changed accordingly.
Best regards

File: src/Data/URI.php (hunk 1 above)
Comment (nhaagen) [reply src/Data/URI.php:416]: Hi @mjansenDatabay , 
I did, at first, but arrays do work :)
please see [here](https://github.com/ILIAS-eLearning/ILIAS/pull/2864/files#diff-71a136a2582520363a4a814f20782245R945-R948).

**Pull request merged**